opentracing-api/src/main/java/io/opentracing/ActiveSpanSource.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+/**
+ * {@link ActiveSpanSource} allows an existing (possibly thread-local-aware) execution context provider to act as a
+ * source for an actively-scheduled OpenTracing Span.
+ *
+ * @see ActiveSpan
+ */
+public interface ActiveSpanSource {
+
+    /**
+     * Return the {@link ActiveSpan active span}. This does not affect the reference count for the {@link ActiveSpan}.
+     *
+     * <p>
+     * If there is an {@link ActiveSpan active span}, it becomes an implicit parent
+     * at {@link Tracer.SpanBuilder#startActive()} time, not at {@link Tracer#buildSpan(String)} time.
+     *
+     * @return the {@link ActiveSpan active span}, or null if none could be found.
+     */
+    ActiveSpan activeSpan();
```

Comment:
@felixbarny this has already been discussed, yes. Unfortunately the github thread here is so convoluted that I can't even find the relevant comment. :-(

I am :100:% happy to add an issue about this (and block release of a new version of OT-Java accordingly), but IMO it can be decoupled from this PR. Just to prep you for my point-of-view: I generally think instrumentation will either (a) create a new Span, in which case there's no need for a `null` check regardless, or (b) be at the layer of a logging library or similar where the verbosity of a `null` check is fine.

If we actually think that it's going to be commonplace to use the OT interface as *the* logging API (rather than a registrant of *the* logging API), then I would be in favor of what you propose. That seems like the essential question.

(I'm less convinced about arguments involving tags: they are almost always set at start()-time in my experience)